cmd/gather/README.md

```diff
@@ -0,0 +1,29 @@
+# OpenTelemetry Operator Must-Gather
```

Comment:
In general, is this utility intended to only be used on OpenShift? The issue makes it sound like it's more generic, but this README specifically restricts it to OpenShift.

Reply:
You can use it without OpenShift:
> Note that you can use this utility too to gather information about the objects deployed by the OpenTelemetry Operator if you don't use OpenShift.

Reply:
That should be closer to the top of the document in my opinion. If you only read the opening paragraph, you walk away with the impression that it's only for OpenShift, and won't read to the end to learn otherwise.